src/include/platform/BuildTime.h

```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+// Adapted from https://stackoverflow.com/questions/11697820/how-to-use-date-and-time-predefined-macros-in-as-two-integers-then-stri
+
+#pragma once
+
+#include <platform/CHIPDeviceConfig.h>
+
+// Expects input as provided by __DATE__ and __TIME__ macros.
+//
+// Example of __DATE__ string: "Jul 27 2012"
```

Comment:
Would the 5th of July be "Jul 05" or "Jul 5" or "Jul  5"?  Should document so the logic below is easier to follow.

Reply:
The format is leading whitespace.  I'll add comments.

Reply:
Added.